openlibrary/plugins/upstream/mybooks.py

```diff
@@ -157,9 +157,9 @@ def GET(self, username):
 
 
 class readinglog_stats(delegate.page):
-    path = "/people/([^/]+)/books/(want-to-read|currently-reading|already-read)/stats"
+    path = "/people/([^/]+)/books/(want-to-read|currently-reading|already-read)/stats(/\\d{4})?"
```

Comment:
Small change in requirements: looking at some of our other APIs, it seems like this pattern would be more consistent, eg https://openlibrary.org/people/ScarTissue/books/already-read/year/2025 . This way adding `/stats` at the end always shows the stats.

```suggestion
    path = "/people/([^/]+)/books/(want-to-read|currently-reading|already-read)(/year/\\d{4})?/stats"
```